tests/Functional/ContentEventRenderTest.php

```diff
@@ -183,42 +167,511 @@ public function testEventRegistrationDateDescription(): void {
+    $node->set('oe_event_registration_dates', [
+      'value' => $registration_start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+      'end_value' => $registration_end_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+    ])->save();
+    $this->drupalGet($node->toUrl());
+
+    $this->assertRegisterButtonDisabled($registration_content);
+    $this->assertEquals('Registration will open today, 18 February 2020, 02:00.', $registration_info_content->getText());
+
+    // Assert "Registration date" field when registration will start in future.
+    $registration_start_date = (clone $static_time)->modify('+ 1 day');
+    $registration_end_date = (clone $static_time)->modify('+ 10 days');
+    $node->set('oe_event_registration_dates', [
+      'value' => $registration_start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+      'end_value' => $registration_end_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+    ])->save();
+    $this->drupalGet($node->toUrl());
+
+    $this->assertRegisterButtonDisabled($registration_content);
+    $this->assertEquals('Registration will open in 1 day. You can register from 19 February 2020, 01:00, until 28 February 2020, 01:00.', $registration_info_content->getText());
+
+    // Assert "Description summary", "Full text", "Featured media",
+    // "Featured media legend" fields (these fields have to be filled all
+    // together).
+    $node->set('oe_event_description_summary', 'Event description summary');
+    $node->set('body', 'Event full text');
+    $node->set('oe_event_featured_media_legend', 'Event featured media legend');
+    $media_image = $this->createMediaImage('event_featured_media');
+    $node->set('oe_event_featured_media', [$media_image])->save();
+    $this->drupalGet($node->toUrl());
+
+    $description_content = $this->assertSession()->elementExists('css', 'article div div:nth-of-type(3)');
+    $text_featured = new TextFeaturedMediaAssert();
+    $text_featured_expected_values = [
+      'title' => 'Description',
+      'caption' => 'Event featured media legend',
+      'text' => 'Event full text',
+      'image' => [
+        'alt' => 'Alternative text event_featured_media',
+        'src' => 'event_featured_media.png',
+      ],
+    ];
+    $text_featured->assertPattern($text_featured_expected_values, $description_content->getHtml());
+
+    // Assert "Report text" and "Summary for report" fields when event finished.
+    $start_date = (clone $static_time)->modify('- 20 days');
+    $end_date = (clone $static_time)->modify('- 10 days');
+    $node->set('oe_event_report_summary', 'Event report summary');
+    $node->set('oe_event_report_text', 'Event report text');
+    $node->set('oe_event_dates', [
+      'value' => $start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+      'end_value' => $end_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+    ])->save();
```

Comment:
See above, when testing time-related features we must never change the actual event dates, we should change time, run cron and reload the page, instead.

Reply:
Done